src/components/transactions/Repay/CollateralRepayModalContent.tsx

```diff
@@ -199,88 +236,87 @@ export function CollateralRepayModalContent({
           {handleBlocked()}
         </Typography>
       )}
-      <Box
-        sx={{
-          bgcolor: 'background.default',
-          border: '1px solid rgba(56, 61, 81, 0.12)',
-          borderRadius: '4px',
-          padding: '8px 16px',
-          mt: 6,
-        }}
+
+      <TxModalDetails
+        gasLimit={gasLimit}
+        slippageSelector={
+          <ListSlippageButton selectedSlippage={maxSlippage} setSlippage={setMaxSlippage} />
+        }
       >
-        <Row caption={<Trans>Price impact</Trans>} captionVariant="subheader1">
-          <FormattedNumber value={priceImpact} variant="secondary14" percent />
-        </Row>
-        <Row caption={<Trans>Minimum received</Trans>} captionVariant="subheader1" sx={{ mt: 4 }}>
-          <FormattedNumber value={outputAmount} variant="secondary14" symbol={poolReserve.symbol} />
-        </Row>
-        <Typography variant="description" sx={{ mt: 4 }}>
-          <Trans>Max slippage rate</Trans>
-        </Typography>
-        <StyledToggleButtonGroup
-          sx={{ mt: 2 }}
-          value={maxSlippage}
-          onChange={(_e, value) => setMaxSlippage(value)}
-          exclusive
-        >
-          <StyledToggleButton value="0.1" sx={{ minWidth: '74px' }}>
-            <Typography variant="secondary14">0.1%</Typography>
-          </StyledToggleButton>
-          <StyledToggleButton value="0.5" sx={{ minWidth: '74px' }}>
-            <Typography variant="secondary14">0.5%</Typography>
-          </StyledToggleButton>
-          <StyledToggleButton value="1" sx={{ minWidth: '74px' }}>
-            <Typography variant="secondary14">1%</Typography>
-          </StyledToggleButton>
-        </StyledToggleButtonGroup>
-      </Box>
-      <TxModalDetails gasLimit={gasLimit}>
+        <DetailsHFLine
+          visibleHfChange={swapVariant === 'exactOut' ? !!amount : !!inputAmount}
+          healthFactor={user?.healthFactor}
+          futureHealthFactor={hfAfterSwap.toString(10)}
+          loading={loadingSkeleton}
+        />
         <DetailsNumberLineWithSub
-          description={<Trans>Remaining debt</Trans>}
+          description={<Trans>Borrow balance after repay</Trans>}
           futureValue={amountAfterRepay.toString()}
           futureValueUSD={displayAmountAfterRepayInUsd.toString()}
           symbol={symbol}
+          tokenIcon={poolReserve.iconSymbol}
+          loading={loadingSkeleton}
+          hideSymbolSuffix
         />
-        <DetailsHFLine
-          visibleHfChange={!!_amount}
-          healthFactor={user?.healthFactor}
-          futureHealthFactor={hfAfterSwap.toString(10)}
+        <DetailsNumberLineWithSub
+          description={<Trans>Collateral balance after repay</Trans>}
+          futureValue={collateralAmountAfterRepay.toString()}
+          futureValueUSD={collateralAmountAfterRepayUSD.toString()}
+          symbol={tokenToRepayWith.symbol}
+          tokenIcon={tokenToRepayWith.iconSymbol}
+          loading={loadingSkeleton}
```

Comment:
Does it make more sense to have `TxModalDetails` take in a single `loading` prop at that level and then manage the skeleton UI at a single place rather than at each line item?

Reply:
I guess not it it's intended to be split out and used in such a granular way: https://github.com/aave/interface/pull/1219/files#diff-ad42037f4acdf239709c517d39133947f03dd5a26fc988c8123891694b1bfa7eR111